CIP-1694/README.md

```diff
@@ -1,530 +1,874 @@
+* A verification key (Ed25519)
+* A native or Plutus script
 
-1. be **ratified**;
-2. be **dropped** as a result of some higher priority action; or else
-3. will **expire** after a number of epochs.
+The blake2b-224 hash digest of a serialized DRep credential is called the _DRep ID_.
 
-Deposits are returned immediately when either:
+The following new types of certificates will be added for governance:
 
-1. a ratified action is **enacted**;
-1. the action **expires**; or
-1. a ratified action is **dropped**.
+##### DRep registration certificates
 
-Ratification is explained in more detail later in this document.
+DRep registration certificates include:
 
->**Note:**
->This design means that the first governance action of a given type to be ratified will be staged for enactment in an epoch.
+* a DRep ID
+* an anchor
 
-#### Enactment
+An **anchor** is a pair of:
 
-Ratified actions may be enacted at an epoch boundary. During enactment, actions in the staging group for the current epoch are prioritized as follows:
+* a URL to a JSON payload of metadata
+* a hash of the contents of this metadata URL
 
-1. Motion of no-confidence;
-2. New Constitutional Committee or a change to the quorum size;
-3. Updates to the Constitution;
-4. Hard Fork initiation;
-5. Protocol parameter changes;
-6. Treasury withdrawals.
+The structure and format of this metadata is deliberately left open in this CIP.
+The on-chain rules will not check either the URL or the hash.
+Client applications should, however, perform the usual sanity checks when fetching content from the provided URL.
 
-As described above, at most one action of each type may be enacted in any given epoch.
-So a ratified _"Motion of no-confidence"_ action will be enacted first if there is one, and so on.
 
-A successful _"Motion of no-confidence"_, the election of a new Constitutional Committee, or a constitutional change invalidates *all* other **unenacted** governance actions (whether or not they have been ratified),
-_causing them to be immediately **dropped** without ever being enacted_. Deposits for dropped actions will be returned immediately.
+##### DRep retirement certificates
 
-#### Content of a Governance Action
+DRep retirement certificates include:
 
-Every governance action will include the following:
+* a DRep ID
+* the epoch number after which the DRep will retire
 
-* a deposit amount (recorded since the amount is an updatable protocol parameter);
-* a reward address to receive the repaid deposit;
-* a URL to any metadata that is needed to justify the action;
-* a hash of the contents of this metadata URL.
+##### Vote delegation certificates
 
-// TODO: Provide a CBOR specification in the annexe for this new on-chain entity
+Vote delegation certificates include:
 
-Additionally, the action will include the following:
+* the DRep ID to which the stake should be delegated
```

Comment:
Is `Abstain` or ` No Confidence` passed into the dRep ID field of the vote delegation certificate? or are these cases handled separately?

Reply:
Yes, those will also be supplied there.